### PR TITLE
Added updated POSTPREINSTALLs and extension.yaml

### PR DESCRIPTION
**Pull request by adreyfus-stripe**
[X] Updated PREINSTALL.md 
[X] Updated POSTINSTALL.md
[X] Updated extension.yaml 
[X] Moved package.json to functions/

### CODE REVIEW

File: POSTINSTALL.md [hunk 1]
```diff
@@ -1,79 +1,163 @@
...
+          description: "Cool hat"
+        }
+      ]
+    })
+    .then(newInvoiceRef => console.log(`added a new invoice at path ${newInvoiceRef.path}`));
+```
+
+Always add an invoice document from your server -- this ensures that your customer cannot directly manipulate the amount and change the order total.
+
+#### Update Cloud Firestore security rules
```
Comment (adreyfus-stripe) [inline POSTINSTALL.md:102]: This section had some outstanding comments in the gdoc. How does it look? Are there some better docs we can link to better explain security rules?

File: POSTINSTALL.md [hunk 2]
```diff
@@ -1,79 +1,163 @@
...
+### Monitoring
 
-2. Read about the Stripe [invoicing workflow](https://stripe.com/docs/billing/invoices/workflow) to understand how invoices are processed.
+As a best practice, you can [monitor the activity](https://firebase.google.com/docs/extensions/manage-installed-extensions#monitor) of your installed extension, including checks on its health, usage, and logs.
 
-3. Customize the colors and logo on your invoice in your Stripe [branding settings page](https://dashboard.stripe.com/settings/branding) in the dashboard.
+Access the [Stripe dashboard](https://dashboard.stripe.com/) to manage all aspects of your Stripe account.
 
-Enjoy and please send any feedback to dev-feedback@stripe.com!
+Enjoy and please send any feedback to stripe-extension-feedback@googlegroups.com!
```
Comment (adreyfus-stripe) [inline POSTINSTALL.md:163]: Did we want to keep this line with the googlegroups email or is there a better place for it?

File: POSTINSTALL.md [hunk 3]
```diff
@@ -1,79 +1,163 @@
...
+
+#### _(Optional)_ Update Cloud Firestore documents with invoice statuses
+
+You can set up a webhook that updates each Cloud Firestore document with the status of its associated invoice (called the [invoice lifecycle](https://stripe.com/docs/billing/subscriptions/overview#invoice-lifecycle)). The webhook creates invoice-status fields in the document, then it updates the fields whenever the invoice's status updates in the Stripe dashboard (for example, with `paid`, `uncollectable`, etc.).  You can then query Cloud Firestore for this status to use in your app, company dashboards, etc.
+
+Here's how to set up the webhook and configure your extension to use it:
+
+1. To enable the Stripe backend to trigger the webhook, make the extension's `updateInvoice` function public by following [these steps](https://cloud.google.com/functions/docs/securing/managing-access-iam#allowing_unauthenticated_function_invocation) in the Google Cloud console. Here's the official name of the function displayed in the console:
+
+`${function:updateInvoice.name}`
```
Comment (adreyfus-stripe) [inline POSTINSTALL.md:137]: Is this formatted correctly?

File: extension.yaml [hunk 4]
```diff
@@ -1,39 +1,69 @@
-name: create-stripe-invoice
-version: 0.0.1
+name: firestore-invoice-stripe
+version: 0.1.0
 specVersion: v1beta
-displayName: Email an invoice to a customer
+
+displayName: Invoice using Stripe
+description: Collect payments from your customers by emailing invoices.
```
Comment (adreyfus-stripe) [inline extension.yaml:6]: Thoughts on this description I just made up? Where is it used?

File: POSTINSTALL.md (hunk 3 above)
Comment (laurenzlong) [reply POSTINSTALL.md:137]: yep!

**Pull request merged**